server/athenian/api/internal/features/metric_calculator.py

```diff
@@ -712,9 +712,11 @@ def _analyze(
         return out
 
     def _agg(self, samples: np.ndarray, offsets: np.ndarray) -> float:
-        lengths = np.diff(offsets.base)
+        lengths = np.diff(offsets.base).astype(float)
```

Comment:
It seems that casting to float is the only required change

Reply:
we have `np.seterr(divide="raise")` so `np.array([1]) / np.array([0])` crashes
offsets is not strictly increasing so lengths includes zeros if I understood correctly when there is a group selecting no facts and it's not the last one. 
Is this faster maybe ?
```
        lenghts = np.diff(offsets.base)
        reduced_samples = np.add.reduceat(samples, offsets)
        # fix NaNs + reduceat copies the previous value
        vec = np.zeros(reduced_samples.shape, dtype=self.dtype)
        np.divide(reduced_samples, lenghts, where=lenghts != 0, out=vec)
        return vec
```

Reply:
This is perfect 👍

Reply:
pushed this version